.wip/api-sketches/src/2-mcp-entrypoint-basic.ts

```diff
@@ -0,0 +1,97 @@
+import { Hono } from 'hono'
+import { basicAuth } from 'hono/basic-auth'
+
+import { WorkerEntrypoint, env } from 'cloudflare:workers'
+import { WorkersMCP, mcp } from 'dream:code'
+
+/**
+ * =============================================
+ * Stateless MCP server, implemented RPC-style
+ * with decorator-based documentation (no build step)
+ * =============================================
+ */
+@mcp({ name: 'Entrypoint demo', version: '1.0.0', route: '/mcp' })
+class McpEntrypoint extends WorkerEntrypoint {
+  @mcp.tool(`
+    Add two numbers the way only MCP can
+    
+    @param a {number} The first number
```

Comment:
...and now i get why you are like "this is the part that could be inferred from the typescript types in a perfect world future"